tensorflow/core/common_runtime/dml/dml_execution_context.cc

```diff
@@ -372,4 +421,176 @@ void DmlExecutionContextImpl::CloseCommandListAndExecute() {
   OpenCommandList();
 }
 
+DmlGpuEvent DmlExecutionContext::CopyBufferRegion(
+    ID3D12Resource* dst_buffer, uint64_t dst_offset,
+    D3D12_RESOURCE_STATES dst_state, ID3D12Resource* src_buffer,
+    uint64_t src_offset, D3D12_RESOURCE_STATES src_state, uint64_t byte_count) {
+  std::unique_lock<std::mutex> lock(shared_state_->mutex);
+
+  shared_state_->WriteBatch().emplace_back([=]() {
+    shared_state_->impl->CopyBufferRegion(dst_buffer, dst_offset, dst_state,
+                                          src_buffer, src_offset, src_state,
+                                          byte_count);
+  });
+
+  shared_state_->new_function_enqueued.notify_all();
+
+  return shared_state_->next_flush_event;
+}
+
+DmlGpuEvent DmlExecutionContext::FillBufferWithPattern(
+    ID3D12Resource* dst, uint64_t dst_offset, uint64_t dst_size_in_bytes,
+    absl::Span<const uint8_t> value) {
+  std::unique_lock<std::mutex> lock(shared_state_->mutex);
+
+  absl::InlinedVector<uint8_t, 16> value_copy(value.size());
+  std::copy(value.begin(), value.end(), value_copy.begin());
+
+  shared_state_->WriteBatch().emplace_back(
+      [=, value_copy = std::move(value_copy)]() {
+        shared_state_->impl->FillBufferWithPattern(
+            dst, dst_offset, dst_size_in_bytes, value_copy);
+      });
+
+  return shared_state_->next_flush_event;
```

Comment:
Are you missing a `notify_all` in this method?

Reply:
Yep. Also fixed in a pending branch, but it's fairly harmless here: FillBuffer will never be the last call so it can't hurt (there will always be a Device sync -> Flush at the end of a session). I can fix here though.